examples/inertia_relief/inertia_relief_example.cpp

```diff
@@ -0,0 +1,542 @@
+  using StateVecs = std::vector<std::shared_ptr<serac::FiniteElementState>>;
+  using DualVecs = std::vector<std::shared_ptr<serac::FiniteElementDual>>;
+
+  ParaviewWriter(std::unique_ptr<mfem::ParaViewDataCollection> pv_, const StateVecs& states_)
+      : pv(std::move(pv_)), states(states_)
+  {
+  }
+
+  ParaviewWriter(std::unique_ptr<mfem::ParaViewDataCollection> pv_, const StateVecs& states_, const StateVecs& duals_)
+      : pv(std::move(pv_)), states(states_), dual_states(duals_)
+  {
+  }
+
+  void write(int step, double time, const std::vector<serac::FiniteElementState const*>& current_states)
+  {
+    SERAC_MARK_FUNCTION;
+    SLIC_ERROR_ROOT_IF(current_states.size() != states.size(), "wrong number of output states to write");
+
+    for (size_t n = 0; n < states.size(); ++n) {
+      auto& state = states[n];
+      *state = *current_states[n];
+      state->gridFunction();
+    }
+
+    pv->SetCycle(step);
+    pv->SetTime(time);
+    pv->Save();
+  }
+
+ private:
+  std::unique_ptr<mfem::ParaViewDataCollection> pv;
+  StateVecs states;
+  StateVecs dual_states;
+};
+
+auto createParaviewOutput(const mfem::ParMesh& mesh, const std::vector<serac::FiniteElementState const*>& states,
+                          std::string output_name)
+{
+  if (output_name == "") {
+    output_name = "default";
+  }
+
+  ParaviewWriter::StateVecs output_states;
+  for (const auto& s : states) {
+    output_states.push_back(std::make_shared<serac::FiniteElementState>(s->space(), s->name()));
+  }
+
+  auto non_const_mesh = const_cast<mfem::ParMesh*>(&mesh);
+  auto paraview_dc = std::make_unique<mfem::ParaViewDataCollection>(output_name, non_const_mesh);
+  int max_order_in_fields = 0;
+
+  // Find the maximum polynomial order in the physics module's states
+  for (const auto& state : output_states) {
+    paraview_dc->RegisterField(state->name(), &state->gridFunction());
+    max_order_in_fields = std::max(max_order_in_fields, state->space().GetOrder(0));
+  }
+
+  // Set the options for the paraview output files
+  paraview_dc->SetLevelsOfDetail(max_order_in_fields);
+  paraview_dc->SetHighOrderOutput(true);
+  paraview_dc->SetDataFormat(mfem::VTKFormat::BINARY);
+  paraview_dc->SetCompression(true);
+
+  return ParaviewWriter(std::move(paraview_dc), output_states, {});
+}
+
+/* NLMCP of the form
+ * 0 <= x \perp F(x, y) >= 0
+ *              Q(x, y)  = 0
+ * Here, F and x are both 0-dimensional
+ * and   Q(x, y) = [ r(u) + (dc/du)^T l]
+ *                 [-c(u)]
+ *            y  = [ u ]
+ *                 [ l ]
+ * we use the approximate Jacobian
+ *       dQ/dy \approx [ dr/du     (dc/du)^T]
+ *                     [-dc/du        0 ]
+ * we note that the sign-convention with regard to "c" is important
+ * as  the approximate Jacobian is positive semi-definite when dr/du is
+ * and thus the NLMC problem is guaranteed to be semi-monotone.
+ */
+class InertialReliefProblem : public GeneralNLMCProblem {
+ protected:
+  mfem::HypreParMatrix* dFdx;
```

Comment:
This is a lot of protected member variables. Should these not be in a private section? Also per our coding guidelines, member variables should end in `_`.